contracts/implementations/MultipleMinter.sol

```diff
@@ -146,4 +146,120 @@ contract MultipleMinter is
+     * The vehicle owner signs a typed structured (EIP-712) message in advance and submits to be verified
+     * @dev Caller must have the minter role
+     * @param data Input data with the following fields:
+     *  manufacturerNode -> Parent manufacturer node id of the vehicle
+     *  owner -> The new nodes owner
+     *  deviceDefinitionId -> The Device Definition Id
+     *  integrationNode -> Parent integration node id of the synthetic device
+     *  vehicleOwnerSig -> Vehicle owner signature hash
+     *  syntheticDeviceSig -> Synthetic Device's signature hash
+     *  syntheticDeviceAddr -> Address associated with the synthetic device
+     *  attrInfoPairsDevice -> List of attribute-info pairs to be added of the synthetic device
+     */
+    function mintVehicleAndSdWithDeviceDefinitionSign(
+        MintVehicleAndSdWithDdInput calldata data
+    ) external onlyRole(MINT_VEHICLE_SD_ROLE) {
+        NodesStorage.Storage storage ns = NodesStorage.getStorage();
+        MapperStorage.Storage storage ms = MapperStorage.getStorage();
+        VehicleStorage.Storage storage vs = VehicleStorage.getStorage();
+        SyntheticDeviceStorage.Storage storage sds = SyntheticDeviceStorage
+            .getStorage();
+
+        address vehicleIdProxyAddress = vs.idProxyAddress;
+        address sdIdProxyAddress = sds.idProxyAddress;
+
+        if (
+            !INFT(IntegrationStorage.getStorage().idProxyAddress).exists(
+                data.integrationNode
+            )
+        ) revert InvalidParentNode(data.integrationNode);
+        if (
+            !INFT(ManufacturerStorage.getStorage().idProxyAddress).exists(
+                data.manufacturerNode
+            )
+        ) revert InvalidParentNode(data.manufacturerNode);
+        if (sds.deviceAddressToNodeId[data.syntheticDeviceAddr] != 0)
+            revert DeviceAlreadyRegistered(data.syntheticDeviceAddr);
+
+        bytes32 message = keccak256(
+            abi.encode(MINT_VEHICLE_SD_TYPEHASH, data.integrationNode)
+        );
+
+        if (
+            !Eip712CheckerInternal._verifySignature(
+                data.syntheticDeviceAddr,
+                message,
+                data.syntheticDeviceSig
+            )
+        ) revert InvalidSdSignature();
+
+        // ----- START Vehicle mint -----
+        uint256 newTokenIdVehicle = INFT(vehicleIdProxyAddress).safeMint(
+            data.owner
+        );
+
+        emit VehicleNodeMintedWithDeviceDefinition(
+            data.manufacturerNode,
+            newTokenIdVehicle,
+            data.owner,
+            data.deviceDefinitionId
+        );
+
+        message = keccak256(
+            abi.encode(
+                MINT_VEHICLE_WITH_DD_TYPEHASH,
+                data.manufacturerNode,
+                data.owner,
+                keccak256(bytes(data.deviceDefinitionId))
+            )
+        );
+
+        if (
+            !Eip712CheckerInternal._verifySignature(
```

Comment:
I'm curious—why do we verify second?

Reply:
No particular reason. I was just following the other function order.